diff --git a/node_modules/@capire/bookshop/srv/admin-service.cds b/node_modules/@capire/bookshop/srv/admin-service.cds
index 883fd1a..a7e4c5d 100644
--- a/node_modules/@capire/bookshop/srv/admin-service.cds
+++ b/node_modules/@capire/bookshop/srv/admin-service.cds
@@ -1,10 +1,20 @@
-using { sap.capire.bookshop as my } from '../db/schema';
+using {sap.capire.bookshop as my} from '../db/schema';
 
-service AdminService @(odata:'/admin') {
+service AdminService @(odata: '/admin') {
   entity Authors as projection on my.Authors;
-  entity Books as projection on my.Books;
-  entity Genres as projection on my.Genres;
+
+  @UI.DeleteHidden: {$edmJson: {$Path: '/AuthorizationRestrictions/isDeleteForbidden'}}
+  entity Books   as projection on my.Books;
+
+  entity Genres  as projection on my.Genres;
+
+  @odata.singleton
+  entity AuthorizationRestrictions {
+    key id                : String;
+        isDeleteForbidden : Boolean default true;
+  }
+
 }
 
-// Additionally serve via HCQL and REST
-annotate AdminService with @hcql @rest;
+// Additionally serve via REST
+annotate AdminService with @rest;
diff --git a/node_modules/@capire/bookshop/srv/admin-service.js b/node_modules/@capire/bookshop/srv/admin-service.js
index f3e910f..b74ce6b 100644
--- a/node_modules/@capire/bookshop/srv/admin-service.js
+++ b/node_modules/@capire/bookshop/srv/admin-service.js
@@ -3,6 +3,13 @@ const cds = require('@sap/cds')
 module.exports = class AdminService extends cds.ApplicationService { init(){
   this.before (['NEW','CREATE'],'Authors', genid)
   this.before (['NEW','CREATE'],'Books', genid)
+
+  this.on("READ", "AuthorizationRestrictions", async (req) => {
+    return {
+      "id": "SINGLETON",
+      "isDeleteForbidden": false
+    }
+  });
   return super.init()
 }}
 
@@ -12,3 +19,4 @@ async function genid (req) {
   const {id} = await SELECT.one.from(req.target).columns('max(ID) as id')
   req.data.ID = id + 4 // Note: that is not safe! ok for this sample only.
 }
+
diff --git a/node_modules/@capire/bookshop/srv/cat-service.cds b/node_modules/@capire/bookshop/srv/cat-service.cds
index b9da4b6..5c4e8fc 100644
--- a/node_modules/@capire/bookshop/srv/cat-service.cds
+++ b/node_modules/@capire/bookshop/srv/cat-service.cds
@@ -19,5 +19,5 @@ service CatalogService @(odata:'/browse') {
   action submitOrder ( book: Books:ID, quantity: Integer );
 }
 
-// Additionally serve via HCQL and REST
-annotate CatalogService with @hcql @rest;
+// Additionally serve via REST
+annotate CatalogService with @rest;
diff --git a/node_modules/@capire/bookshop/srv/cat-service.js b/node_modules/@capire/bookshop/srv/cat-service.js
index 1598f05..2758077 100644
--- a/node_modules/@capire/bookshop/srv/cat-service.js
+++ b/node_modules/@capire/bookshop/srv/cat-service.js
@@ -2,7 +2,6 @@ const cds = require('@sap/cds')
 
 class CatalogService extends cds.ApplicationService { init() {
 
-  const { Books } = cds.entities ('sap.capire.bookshop')
   const { ListOfBooks } = this.entities
 
   // Add some discount for overstocked books
@@ -14,11 +13,11 @@ class CatalogService extends cds.ApplicationService { init() {
   this.on ('submitOrder', async req => {
     let { book:id, quantity } = req.data
     if (quantity < 1) return req.error (400, `quantity has to be 1 or more`)
-    let succeeded = await UPDATE (Books,id) 
+    let succeeded = await UPDATE ('sap.capire.bookshop.Books',id)
       .with `stock = stock - ${quantity}`
       .where `stock >= ${quantity}`
     if (succeeded) return
-    else if (!this.exists(Books,id)) req.error (404, `Book #${id} doesn't exist`)
+    else if (!await this.exists('sap.capire.bookshop.Books',id)) req.error (404, `Book #${id} doesn't exist`)
     else req.error (409, `${quantity} exceeds stock for book #${id}`)
   })
 
